services/director-v2/src/simcore_service_director_v2/models/schemas/dynamic_services/scheduler.py

```diff
@@ -432,42 +370,16 @@ def from_http_request(
         return cls.parse_obj(obj_dict)
 
     @classmethod
-    def from_service_labels_stored_data(
-        cls,
-        service_labels_stored_data: ServiceLabelsStoredData,
-        port: Optional[int],
-        request_dns: str = None,
-        request_scheme: str = None,
-        proxy_service_name: str = None,
+    def from_service_inspect(
+        cls, service_inspect: Mapping[str, Any]
     ) -> "SchedulerData":
-        return cls.parse_obj(
-            dict(
-                service_name=service_labels_stored_data.service_name,
-                node_uuid=service_labels_stored_data.node_uuid,
-                project_id=service_labels_stored_data.project_id,
-                user_id=service_labels_stored_data.user_id,
-                key=service_labels_stored_data.key,
-                version=service_labels_stored_data.version,
-                paths_mapping=service_labels_stored_data.paths_mapping,
-                compose_spec=json.dumps(service_labels_stored_data.compose_spec),
-                container_http_entry=service_labels_stored_data.container_http_entry,
-                restart_policy=service_labels_stored_data.restart_policy,
-                dynamic_sidecar_network_name=service_labels_stored_data.dynamic_sidecar_network_name,
-                simcore_traefik_zone=service_labels_stored_data.simcore_traefik_zone,
-                service_port=service_labels_stored_data.service_port,
-                request_dns=request_dns,
-                request_scheme=request_scheme,
-                proxy_service_name=proxy_service_name,
-                dynamic_sidecar=dict(
-                    hostname=service_labels_stored_data.service_name,
-                    port=port,
-                ),
-            )
-        )
+        labels = service_inspect["Spec"]["Labels"]
+        return cls.parse_obj(json.loads(labels[DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL]))
```

Comment:
TIP: you can use ``cls.parse_raw(labels[DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL])`` here